### PR TITLE
feat: set target-type for ALBs to ip

### PR DESCRIPTION
**Pull request by kaldorn**
On the tin.

Adds the following annotation:
```
        'alb.ingress.kubernetes.io/target-type': 'ip',
```

Any other changes are from `jsonnetfmt` running.

### CODE REVIEW

**Review by dvagapov — APPROVED** [review]

**Comment by dvagapov** [comment]
From https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/ingress/annotations/#target-type -> 
```
network plugin must use secondary IP addresses on [ENI](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html) for pod IP to use ip mode. e.g.
[amazon-vpc-cni-k8s](https://github.com/aws/amazon-vpc-cni-k8s)

ip mode is required for sticky sessions to work with Application Load Balancers.
```

File: kubernetes/outreach.libsonnet [hunk 1]
```diff
@@ -225,23 +225,24 @@ k + kubecfg {
...
+        'alb.ingress.kubernetes.io/actions.ssl-redirect': '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}',  // Redirect http to https
         'alb.ingress.kubernetes.io/ssl-policy': 'ELBSecurityPolicy-TLS13-1-2-Res-2021-06',
         'alb.ingress.kubernetes.io/scheme': scheme,
-        'alb.ingress.kubernetes.io/load-balancer-attributes': 'routing.http.drop_invalid_header_fields.enabled=true,access_logs.s3.enabled=true,access_logs.s3.bucket=outreach-aws-lb-controller-logs-%s,access_logs.s3.prefix=%s,idle_timeout.timeout_seconds=%s' % [cluster.region, groupName, idleTimeoutSeconds], 
+        'alb.ingress.kubernetes.io/load-balancer-attributes': 'routing.http.drop_invalid_header_fields.enabled=true,access_logs.s3.enabled=true,access_logs.s3.bucket=outreach-aws-lb-controller-logs-%s,access_logs.s3.prefix=%s,idle_timeout.timeout_seconds=%s' % [cluster.region, groupName, idleTimeoutSeconds],
         'alb.ingress.kubernetes.io/success-codes': '200-399',
+        'alb.ingress.kubernetes.io/target-type': 'ip',
```
Comment (zdenekvicar) [inline kubernetes/outreach.libsonnet:239]: Hello @kaldorn , based on previous incident caused by changes in jsonnet-libs [ICM-676](https://outreach-io.atlassian.net/browse/ICM-676), I need to ask for a condition to ensure this is not spreading everywhere without a control. One of the learnings in the incident was that we need to ensure changes are deployed on staging where we need a chance to see how this works and whether it affects anything. Could you please add such condition ? 

Also, I am missing some context in the PR. Why are we doing this change ? What is the benefit of doing this, and is there any downside of this ? Is this part of any bigger project (like the CrossAZ networking which @jan-ludvik is leading) ?

I am blocking the PR until we have those questions answered and staging condition in place.

**Review by zdenekvicar — CHANGES_REQUESTED** [review]

File: kubernetes/outreach.libsonnet (hunk 1 above)
Comment (zdenekvicar) [reply kubernetes/outreach.libsonnet:239]: discussion moved to slack 👉 https://outreach-hq.slack.com/archives/C06HLD1DARH/p1707388405057919

**Review by jan-ludvik — APPROVED** [review]
Will be deployed only to staging. Awesome, thanks!

**Review by zdenekvicar — APPROVED** [review]
All my questions / concerns were explained and/or addressed in slack thread: https://outreach-hq.slack.com/archives/C06HLD1DARH/p1707388405057919

LGTM, thank you

**Pull request merged**